api/tests/integration/infrastructure/repositories/campaign-participation-repository_test.js

```diff
@@ -162,6 +163,71 @@ describe('Integration | Repository | Campaign Participation', () => {
     });
   });
 
+  describe('markPreviousParticipationAsImproved', () => {
+    it('marks previous participation as improved', async () => {
+      const userId = databaseBuilder.factory.buildUser().id;
+      const campaignId = databaseBuilder.factory.buildCampaign({ multipleSendings: true }).id;
+      const oldCampaignParticipationId = databaseBuilder.factory.buildCampaignParticipation({ userId, campaignId, isShared: true, sharedAt: new Date('2020-01-01'), isImproved: false }).id;
+
+      await databaseBuilder.commit();
+
+      await DomainTransaction.execute((domainTransaction) => {
+        return campaignParticipationRepository.markPreviousParticipationAsImproved(campaignId, userId, domainTransaction);
+      });
+
+      const campaignParticipation = await knex('campaign-participations').where({ id: oldCampaignParticipationId }).first();
+
+      expect(campaignParticipation.isImproved).to.equals(true);
+    });
+
+    it('does not change the campaign participation', async () => {
+      const userId = databaseBuilder.factory.buildUser().id;
+      const campaignId = databaseBuilder.factory.buildCampaign({ multipleSendings: true }).id;
+      const oldCampaignParticipationId = databaseBuilder.factory.buildCampaignParticipation({ userId, campaignId, isShared: true, sharedAt: new Date('2020-01-01'), isImproved: false }).id;
+
+      await databaseBuilder.commit();
+      try {
+        await DomainTransaction.execute(async (domainTransaction) => {
+          await campaignParticipationRepository.markPreviousParticipationAsImproved(campaignId, userId, domainTransaction);
+          throw new Error();
+        });
+        // eslint-disable-next-line no-empty
+      } catch {}
```

Comment:
**question** : on est obligés de faire ça comme ça ?

Reply:
Oui on test le fait que la transaction peut être rollback le it n'est pas bon

Reply:
c'est le test qui n'est pas bien nommé

Reply:
done